### PR TITLE
Move to maven and GitHub actions

### PR DESCRIPTION
**Pull request by MancunianSam**
- Move to GitHub actions
- Deploy to maven
Needs https://github.com/nationalarchives/tdr-github-actions/pull/12 to be merged.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: TDR Deploy Auth Utils
+on:
+  workflow_dispatch:
+
```
Comment (TomJKing) [inline .github/workflows/deploy.yml:4]: Is this line break needed?

File: build.sbt [hunk 2]
```diff
@@ -17,7 +17,7 @@ ThisBuild / developers := List(
     id    = "SP",
```
Comment (TomJKing) [inline build.sbt:17]: Should this be your details? Have we got a generic TDR account that could be used instead?

**Review by TomJKing — COMMENTED** [review]
Looks good. Couple of minor comments. 

I moving to Maven, from s3 storage is a big enough change to warrant an ADR, what do you reckon?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: Deploy if not a version bump PR
+on:
+  push:
+    branches:
+      - master
+      - move-to-maven-and-github-actions
```
Comment (TomJKing) [inline .github/workflows/build.yml:6]: Should this branch name be here?

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (MancunianSam) [reply .github/workflows/deploy.yml]: No, I've taken it out.

File: build.sbt (hunk 2 above)
Comment (MancunianSam) [reply build.sbt]: I've put the Jenkins GitHub user in there now.

File: .github/workflows/build.yml (hunk 3 above)
Comment (MancunianSam) [reply .github/workflows/build.yml]: Nope, I took it out and forgot to commit it.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**